src/lib/storage/storage_manager.cpp

```diff
@@ -7,45 +7,57 @@
 
 #include "utils/assert.hpp"
 
+#include <boost/range/combine.hpp>
+
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _table_names.push_back(name);
+  _table_ptrs.push_back(table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto table_index = get_table_index(name);
+  _table_names.erase(_table_names.begin() + table_index);
+  _table_ptrs.erase(_table_ptrs.begin() + table_index);
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  return nullptr;
+  return _table_ptrs.at(get_table_index(name));
 }
 
 bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  return false;
+  return std::find(_table_names.begin(), _table_names.end(), name) != _table_names.end();
 }
 
-std::vector<std::string> StorageManager::table_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
+std::vector<std::string> StorageManager::table_names() const { return _table_names; }
 
 void StorageManager::print(std::ostream& out) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
-
-void StorageManager::reset() {
-  // Implementation goes here
+  for (auto const& [table_name, table_ptr] : boost::combine(_table_names, _table_ptrs)) {
+    out << "=== " << table_name << " ===" << std::endl;
+    out << "n columns: " << table_ptr->column_count() << std::endl;
+    out << "n rows: " << table_ptr->row_count() << std::endl;
+    out << "n chunks: " << table_ptr->chunk_count() << std::endl;
+    out << "columns:" << std::endl;
+    for (ColumnID col_id = ColumnID{0}; col_id < table_ptr->column_count(); col_id++) {
+      out << "  " << table_ptr->column_name(col_id) << " (" << table_ptr->column_type(col_id) << ")" << std::endl;
+    }
+  }
+}
+
+void StorageManager::reset() { get() = StorageManager{}; }
+
+uint16_t StorageManager::get_table_index(const std::string& name) const {
+  auto offset = std::find(_table_names.begin(), _table_names.end(), name);
+  if (offset == _table_names.end()) {
+    throw std::invalid_argument("table name is unknown");
+  }
```

Comment:
This could get quite unhandy for real-world databases with > 100,000 tables (such as ERP systems). Thus, us a hash map. Also, the data type is not sufficient for these cases.